packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -58,8 +58,10 @@ export function getAffectedGlobModules(
           (!affirmed.length || affirmed.some((glob) => isMatch(file, glob))) &&
           (!negated.length || negated.every((glob) => isMatch(file, glob))),
       )
-    )
-      modules.push(...(server.moduleGraph.getModulesByFile(id) || []))
+    ) {
+      const mod = server.moduleGraph.getModuleById(id)
+      if (mod) modules.push(mod)
+    }
```

Comment:
An id (e.g. `package/foo/index.js?v=version`) is set here and `server.moduleGraph.getModulesByFile(id)` was returning `[]`.
https://github.com/vitejs/vite/blob/9b7b4ed99ff30ca8eed520233b89e5f977015656/packages/vite/src/node/plugins/importMetaGlob.ts#L93-L94

I guess it was wrong to use `server.moduleGraph.getModuleById` as the `_importGlobMap` uses an `id` instead of `file`.

Another way to fix this is to remove `?v=version` before calling `getModulesByFile` or setting `_importGlobMap`.